src/main/java/sparksoniq/JsoniqQueryExecutor.java

```diff
@@ -255,7 +255,7 @@ private String getIteratorOutput(RuntimeIterator iterator) {
             result = iterator.next();
         }
         if (result == null) {
-            return "";
+            return iterator.getResultIfEmpty().serialize();
```

Comment:
If result == null, then it means the iterator returned the empty sequence, which should always be serialized as an empty string.

You probably had to make this change because of a (pre-existing) mistake in the implementation of instance of : if the left hand side is (), then false should be returned by the iterator and not the empty sequence.

The current implementation of the iterator probably incorrectly returns () instead (see comment in the instance of iterator code). If you fix the iterator, the getResultIfEmpty function is no longer necessary.

Reply:
_I will respond here to this and the following comments about the issue with the empty list._

From the second revision, i.e. commit 042b902, I removed the function _getResultIfEmpty()_ since, as @ghislainfourny rightfully said, it was superfluous. I had to change the behavior in the _open()_ function of the parent UnaryOperationIterator because it was setting **this._hasNext** to false when **child.hasNext()** was false (and EmptySequenceIterator always return false for hasNext() ). Now the check for __child.hasNext()_ happens in the _next()_ function for both UnaryOperationIterator and InstanceOfIterator.